website/addons/base/views.py

```diff
@@ -435,6 +435,7 @@ def addon_view_file(auth, node, node_addon, guid_file, extras):
         # Note: must be called after get_or_start_render. This is really only for github
         'size': size,
         'extra': json.dumps(getattr(guid_file, 'extra', {})),
+        'source_url': getattr(guid_file, 'source_url', None),
```

Comment:
Generally, I'd only use `getattr()` if the attribute I was attempting to access wasn't known ahead of time. 

Have you considered setting `GuidFile.source_url` at the level on the parent class, and defaulting it to `None`? That would eliminate the need for this logic and ensure that if this property is needed by a sibling class that it doesn't get implemented a second time.